app/src/main/java/com/example/caloriecounter/feature/restaurant/ui/meals/MealListScreen.kt

```diff
@@ -0,0 +1,87 @@
+package com.example.caloriecounter.feature.restaurant.ui.meals
+
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.fadeIn
+import androidx.compose.animation.fadeOut
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.CircularProgressIndicator
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.example.caloriecounter.R
+import com.example.caloriecounter.ui.ErrorScreen
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+@Composable
+fun MealListScreen(
+    mealsState: MealsContract.State,
+    effectFlow: Flow<MealsContract.Effect>,
+    onEventSent: (event: MealsContract.Event) -> Unit,
+    onNavigationRequested: (mealId: String) -> Unit,
+) {
+    LaunchedEffect(Unit) {
+        effectFlow.onEach { effect ->
+            when (effect) {
+                is MealsContract.Effect.Navigation.ToMealRecipe -> {
+                    onNavigationRequested(effect.mealId)
+                }
+            }
+        }.launchIn(this)
+    }
+
+    Box(modifier = Modifier.fillMaxSize()) {
+        Column {
+            when (mealsState) {
+                is MealsContract.State.Success -> {
+                    Text(
+                        text = mealsState.category,
+                        modifier = Modifier.fillMaxWidth(),
+                        style = MaterialTheme.typography.h5,
+                        textAlign = TextAlign.Center,
+                        color = MaterialTheme.colors.primary
```

Comment:
In case you will reuse this kind of Text with the same style, textAlign and maybe color it's better to extract it to utils functions. In our app we have a `@Composable` for all basic Typography `Headtitle, subtitle, H1..H5` and so on. Maybe it will also help you

Reply:
If I use this component again, I will create a new `@Composable`.  I don't think making so many functions for a small app will be a good idea, especially when these functions are not used.

Reply:
Okay, thanks for the answer